segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -164,8 +169,18 @@ public PravegaRequestProcessor(StreamSegmentStore segmentStore, ServerConnection
     public void readSegment(ReadSegment readSegment) {
         Timer timer = new Timer();
         final String segment = readSegment.getSegment();
+
+        if (this.tokenVerifier != null && !tokenVerifier.verifyToken(segment,
+                readSegment.getDelegationToken(), PravegaAuthHandler.PravegaAccessControlEnum.READ)) {
+            log.warn("Delegation token verification failed");
+            handleException(readSegment.getOffset(), segment,
+                    "Seal Segment", new PravegaAuthenticationException("Token verification failed"));
```

Comment:
Please make sure these messages match the name of the operation. This should be "Read Segment".

Please verify below as well.

Reply:
Fixed.